pkg/skaffold/runner/deployer.go

```diff
@@ -86,13 +90,28 @@ func GetDeployer(runCtx *v2.RunContext, provider deploy.ComponentProvider, label
 			deployers = append(deployers, deployer)
 		}
 
+		// TODO: deprecate KustomizeDeploy
 		if d.KustomizeDeploy != nil {
 			deployer, err := kustomize.NewDeployer(dCtx, labels, provider, d.KustomizeDeploy)
 			if err != nil {
 				return nil, err
 			}
 			deployers = append(deployers, deployer)
 		}
+
+		if d.KptV2Deploy != nil {
+			if d.KptV2Deploy.Dir == "" {
+				logrus.Debugln("use default render directory as deployment directory.")
+				hydrationDir, err := runCtx.GetRenderOutputPath()
+				if err != nil {
+					return nil, err
+				}
+				logrus.Infof("manifests are deployed from render path %v\n", hydrationDir)
```

Comment:
move this outside of the `if d.KptV2Deploy.Dir == ""` check

Reply:
done.